### ISSUE TITLE
Crash with 6.1.BETA10 clicking recipe for Thaumometer in Thaumonomicon

### ISSUE THREAD

**Issue by hanleybrand** [Azanor/thaumcraft-beta#512]
I was just trying the mod out (glad to see it being active again, btw) and I got a CTD when trying to view the recipe for the thaumometer in the Thaumonomicon. 

I made a new forge/thaumcraft install, installing only JEI and Thaumic JEI and I still got the crash. 

Then I removed  JEI and Thaumic JEI and I still get the crash.  Crashlog from the Thaumcraft/Baubles only instance is below. 

```
---- Minecraft Crash Report ----
// Don't be sad, have a hug! <3

Time: 4/26/18 9:55 AM
Description: Updating screen events

java.lang.NullPointerException: Updating screen events
	at thaumcraft.client.gui.GuiResearchPage.func_73864_a(GuiResearchPage.java:1966)
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:533)
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:501)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1757)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1096)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:397)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at thaumcraft.client.gui.GuiResearchPage.func_73864_a(GuiResearchPage.java:1966)
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:533)
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:501)

-- Affected screen --
Details:
	Screen name: thaumcraft.client.gui.GuiResearchPage

-- Affected level --
Details:
	Level name: MpServer
	All players: 1 total; [EntityPlayerSP['hanleybrand'/266, l='MpServer', x=48.15, y=68.00, z=76.81]]
	Chunk stats: MultiplayerChunkCache: 564, 564
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: World: (228,64,256), Chunk: (at 4,4,0 in 14,16; contains blocks 224,0,256 to 239,255,271), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 20791 game time, 27869 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Forced entities: 148 total; [EntityItem['item.item.rottenFlesh'/0, l='MpServer', x=45.42, y=69.00, z=67.26], EntityBrainyZombie['Angry Zombie'/1024, l='MpServer', x=83.50, y=22.00, z=88.50], EntityCreeper['Creeper'/2, l='MpServer', x=33.54, y=41.00, z=111.66], EntityItem['item.item.crystal_essence'/3, l='MpServer', x=37.38, y=26.00, z=130.88], EntityItem['item.item.crystal_essence'/4, l='MpServer', x=37.27, y=26.00, z=129.40], EntityItem['item.item.crystal_essence'/5, l='MpServer', x=38.42, y=26.00, z=129.65], EntityBat['Bat'/773, l='MpServer', x=86.68, y=49.29, z=73.56], EntityCreeper['Creeper'/263, l='MpServer', x=70.50, y=22.00, z=57.50], EntityVillager['Villager'/8, l='MpServer', x=46.50, y=68.00, z=75.70], EntityBat['Bat'/2056, l='MpServer', x=3.15, y=18.47, z=121.17], EntityItem['item.item.rottenFlesh'/9, l='MpServer', x=51.67, y=68.00, z=71.13], EntityZombie['Zombie'/265, l='MpServer', x=103.50, y=38.00, z=57.50], EntityBat['Bat'/2057, l='MpServer', x=8.87, y=17.04, z=127.65], EntityItem['item.item.rottenFlesh'/10, l='MpServer', x=53.26, y=69.50, z=71.63], EntityBat['Bat'/11, l='MpServer', x=56.42, y=41.11, z=82.66], EntityBat['Bat'/2061, l='MpServer', x=9.11, y=17.75, z=127.16], EntityPlayerSP['hanleybrand'/266, l='MpServer', x=48.15, y=68.00, z=76.81], EntitySkeleton['Skeleton'/1049, l='MpServer', x=67.50, y=44.00, z=26.50], EntityVillager['Villager'/27, l='MpServer', x=71.71, y=70.00, z=74.50], EntityVillager['Villager'/28, l='MpServer', x=43.61, y=67.94, z=43.53], EntityVillager['Villager'/29, l='MpServer', x=68.48, y=70.00, z=75.41], EntityCreeper['Creeper'/47, l='MpServer', x=109.20, y=36.00, z=105.50], EntityCreeper['Creeper'/48, l='MpServer', x=102.50, y=37.00, z=105.50], EntityCow['Cow'/51, l='MpServer', x=109.51, y=71.00, z=113.18], EntityCow['Cow'/310, l='MpServer', x=-29.38, y=68.00, z=86.21], EntityCow['Cow'/311, l='MpServer', x=-28.82, y=69.00, z=76.77], EntityBat['Bat'/1594, l='MpServer', x=15.95, y=16.46, z=118.47], EntityBat['Bat'/1341, l='MpServer', x=-31.52, y=20.10, z=15.25], EntitySkeleton['Skeleton'/2115, l='MpServer', x=18.12, y=29.00, z=66.50], EntityCow['Cow'/68, l='MpServer', x=112.50, y=68.00, z=101.50], EntityCow['Cow'/324, l='MpServer', x=-18.30, y=69.00, z=141.15], EntityCow['Cow'/69, l='MpServer', x=113.53, y=68.00, z=100.83], EntityZombie['Zombie'/325, l='MpServer', x=-1.50, y=43.00, z=109.21], EntityCow['Cow'/70, l='MpServer', x=117.18, y=72.00, z=116.20], EntityChicken['Chicken'/71, l='MpServer', x=121.20, y=86.00, z=129.54], EntityZombie['Zombie'/839, l='MpServer', x=33.50, y=28.00, z=125.50], EntityChicken['Chicken'/72, l='MpServer', x=118.54, y=84.00, z=127.12], EntityCow['Cow'/328, l='MpServer', x=-14.94, y=69.00, z=146.37], EntityChicken['Chicken'/73, l='MpServer', x=118.50, y=88.00, z=131.50], EntityCow['Cow'/329, l='MpServer', x=-13.50, y=69.00, z=147.50], EntityChicken['Chicken'/74, l='MpServer', x=118.44, y=87.00, z=129.70], EntityCow['Cow'/330, l='MpServer', x=-2.58, y=70.00, z=156.76], EntityCow['Cow'/331, l='MpServer', x=-3.84, y=69.00, z=154.50], EntityCow['Cow'/333, l='MpServer', x=-17.23, y=69.00, z=148.41], EntityCow['Cow'/334, l='MpServer', x=-16.36, y=69.00, z=149.43], EntityItem['item.item.crystal_essence'/336, l='MpServer', x=-1.48, y=24.00, z=54.40], EntitySkeleton['Skeleton'/1616, l='MpServer', x=-16.50, y=58.00, z=43.50], EntityItem['item.item.crystal_essence'/337, l='MpServer', x=-7.36, y=24.00, z=55.88], EntityZombie['Zombie'/1361, l='MpServer', x=50.50, y=19.00, z=99.50], EntityZombie['Zombie'/1617, l='MpServer', x=-16.50, y=58.00, z=44.50], EntityItem['item.item.crystal_essence'/338, l='MpServer', x=-4.49, y=26.00, z=54.08], EntityZombieVillager['Zombie Villager'/1362, l='MpServer', x=59.50, y=40.00, z=32.50], EntityItem['item.item.crystal_essence'/339, l='MpServer', x=-0.83, y=24.00, z=53.21], EntitySheep['Sheep'/342, l='MpServer', x=9.50, y=66.00, z=3.50], EntitySheep['Sheep'/343, l='MpServer', x=3.78, y=66.00, z=0.58], EntityZombie['Zombie'/344, l='MpServer', x=-28.50, y=53.00, z=43.50], EntitySheep['Sheep'/345, l='MpServer', x=15.30, y=65.00, z=-2.24], EntityZombie['Zombie'/347, l='MpServer', x=6.78, y=29.00, z=73.45], EntityPig['Pig'/348, l='MpServer', x=10.50, y=64.00, z=70.50], EntityPig['Pig'/349, l='MpServer', x=14.50, y=64.00, z=70.50], EntityCow['Cow'/350, l='MpServer', x=10.34, y=64.00, z=74.17], EntitySkeleton['Skeleton'/2142, l='MpServer', x=49.50, y=21.00, z=111.50], EntityCow['Cow'/351, l='MpServer', x=10.11, y=64.00, z=76.73], EntitySkeleton['Skeleton'/2143, l='MpServer', x=49.50, y=21.00, z=108.50], EntityCow['Cow'/352, l='MpServer', x=1.87, y=64.00, z=67.86], EntityCreeper['Creeper'/1120, l='MpServer', x=4.50, y=28.00, z=31.50], EntitySkeleton['Skeleton'/2144, l='MpServer', x=48.50, y=21.00, z=108.50], EntityCow['Cow'/353, l='MpServer', x=13.18, y=64.00, z=73.74], EntitySkeleton['Skeleton'/1889, l='MpServer', x=-3.50, y=41.00, z=127.50], EntityCreeper['Creeper'/2145, l='MpServer', x=48.50, y=21.00, z=107.50], EntitySkeleton['Skeleton'/354, l='MpServer', x=12.53, y=67.00, z=64.25], EntitySkeleton['Skeleton'/1122, l='MpServer', x=37.50, y=23.00, z=98.50], EntitySheep['Sheep'/355, l='MpServer', x=10.52, y=64.00, z=79.88], EntityZombie['Zombie'/1379, l='MpServer', x=9.50, y=44.00, z=94.80], EntityZombie['Zombie'/2147, l='MpServer', x=5.50, y=33.00, z=19.50], EntityPig['Pig'/356, l='MpServer', x=6.61, y=64.00, z=71.80], EntityZombie['Zombie'/1380, l='MpServer', x=7.50, y=43.00, z=95.72], EntitySheep['Sheep'/357, l='MpServer', x=12.17, y=65.00, z=87.24], EntityZombie['Zombie'/1381, l='MpServer', x=2.50, y=42.00, z=104.50], EntityPig['Pig'/358, l='MpServer', x=5.40, y=64.00, z=79.51], EntitySkeleton['Skeleton'/870, l='MpServer', x=5.50, y=36.00, z=76.50], EntitySkeleton['Skeleton'/359, l='MpServer', x=-5.50, y=33.00, z=73.50], EntityCow['Cow'/360, l='MpServer', x=-31.22, y=69.00, z=86.19], EntityZombie['Zombie'/1653, l='MpServer', x=44.50, y=17.00, z=100.50], EntityCreeper['Creeper'/889, l='MpServer', x=-12.50, y=25.00, z=58.50], EntitySkeleton['Skeleton'/1145, l='MpServer', x=108.50, y=17.00, z=90.50], EntityZombie['Zombie'/1146, l='MpServer', x=108.50, y=17.00, z=91.50], EntityCow['Cow'/395, l='MpServer', x=25.23, y=68.00, z=156.53], EntityBat['Bat'/1421, l='MpServer', x=42.52, y=24.58, z=134.52], EntityBat['Bat'/1422, l='MpServer', x=44.60, y=24.58, z=133.67], EntityEnderman['Enderman'/1680, l='MpServer', x=-28.48, y=20.00, z=148.28], EntityVillager['Villager'/402, l='MpServer', x=43.57, y=67.94, z=41.49], EntityVillager['Villager'/403, l='MpServer', x=57.30, y=69.00, z=42.73], EntityPig['Pig'/404, l='MpServer', x=13.16, y=65.00, z=86.97], EntityPig['Pig'/405, l='MpServer', x=12.50, y=64.00, z=85.50], EntityBat['Bat'/1941, l='MpServer', x=49.62, y=20.15, z=98.38], EntitySheep['Sheep'/406, l='MpServer', x=15.24, y=69.00, z=95.43], EntitySheep['Sheep'/407, l='MpServer', x=14.44, y=64.00, z=84.27], EntitySkeleton['Skeleton'/1177, l='MpServer', x=49.50, y=21.00, z=134.50], EntityPig['Pig'/410, l='MpServer', x=17.51, y=68.00, z=64.75], EntityItem['item.item.crystal_essence'/411, l='MpServer', x=3.78, y=15.00, z=132.00], EntityItem['item.item.crystal_essence'/412, l='MpServer', x=1.13, y=13.00, z=133.13], EntityItem['item.item.crystal_essence'/413, l='MpServer', x=5.14, y=18.00, z=133.64], EntityItem['item.item.seeds'/414, l='MpServer', x=0.59, y=68.00, z=137.27], EntityItem['item.tile.mushroom'/415, l='MpServer', x=24.14, y=12.00, z=112.80], EntitySkeleton['Skeleton'/1445, l='MpServer', x=102.50, y=36.00, z=103.50], EntityPig['Pig'/425, l='MpServer', x=28.78, y=67.00, z=31.51], EntitySheep['Sheep'/432, l='MpServer', x=18.49, y=64.00, z=-0.29], EntityPig['Pig'/433, l='MpServer', x=28.53, y=64.00, z=8.47], EntityPig['Pig'/434, l='MpServer', x=20.50, y=64.00, z=4.35], EntityPig['Pig'/435, l='MpServer', x=20.50, y=64.00, z=5.55], EntityBat['Bat'/1715, l='MpServer', x=61.98, y=24.94, z=151.16], EntityPig['Pig'/436, l='MpServer', x=27.50, y=66.00, z=12.25], EntityPig['Pig'/444, l='MpServer', x=33.23, y=67.00, z=36.80], EntityPig['Pig'/445, l='MpServer', x=34.52, y=67.00, z=34.67], EntityChicken['Chicken'/448, l='MpServer', x=55.50, y=70.00, z=16.50], EntityItem['item.tile.sapling_greatwood'/451, l='MpServer', x=17.57, y=68.00, z=61.66], EntityPig['Pig'/452, l='MpServer', x=27.43, y=68.00, z=49.72], EntityChicken['Chicken'/454, l='MpServer', x=54.50, y=71.00, z=11.50], EntityChicken['Chicken'/455, l='MpServer', x=55.50, y=71.00, z=13.50], EntityPig['Pig'/458, l='MpServer', x=36.20, y=67.00, z=31.66], EntityZombie['Zombie'/1226, l='MpServer', x=79.50, y=26.00, z=67.50], EntitySkeleton['Skeleton'/974, l='MpServer', x=73.51, y=31.00, z=25.68], EntityCreeper['Creeper'/1487, l='MpServer', x=61.50, y=30.00, z=18.50], EntityCreeper['Creeper'/473, l='MpServer', x=-19.50, y=40.00, z=132.50], EntityBat['Bat'/2009, l='MpServer', x=27.46, y=34.37, z=13.49], EntityBat['Bat'/2010, l='MpServer', x=29.16, y=37.30, z=8.72], EntityZombie['Zombie'/1759, l='MpServer', x=109.50, y=36.00, z=107.50], EntitySheep['Sheep'/480, l='MpServer', x=84.41, y=72.00, z=17.25], EntityZombie['Zombie'/1760, l='MpServer', x=110.50, y=36.00, z=104.50], EntitySheep['Sheep'/481, l='MpServer', x=86.76, y=72.00, z=14.50], EntityVillager['Villager'/482, l='MpServer', x=75.52, y=70.94, z=55.66], EntitySheep['Sheep'/484, l='MpServer', x=75.77, y=72.00, z=21.45], EntitySheep['Sheep'/485, l='MpServer', x=74.79, y=72.00, z=16.71], EntityChicken['Chicken'/486, l='MpServer', x=65.17, y=71.00, z=20.50], EntityItem['item.item.seeds'/487, l='MpServer', x=102.41, y=69.00, z=24.88], EntityItem['item.item.seeds'/488, l='MpServer', x=103.09, y=69.00, z=23.70], EntityItem['item.item.seeds'/489, l='MpServer', x=102.32, y=69.00, z=23.95], EntityZombie['Zombie'/1001, l='MpServer', x=74.50, y=19.00, z=108.50], EntityItem['item.item.seeds'/490, l='MpServer', x=106.68, y=70.00, z=28.07], EntityItem['item.item.seeds'/491, l='MpServer', x=104.56, y=69.00, z=19.96], EntityItem['item.item.seeds'/492, l='MpServer', x=104.08, y=69.00, z=21.73], EntityCreeper['Creeper'/2034, l='MpServer', x=102.50, y=33.00, z=81.50], EntityCreeper['Creeper'/2035, l='MpServer', x=104.50, y=33.00, z=85.50], EntitySpider['Spider'/2036, l='MpServer', x=102.50, y=33.00, z=85.50], EntitySkeleton['Skeleton'/2041, l='MpServer', x=119.50, y=53.00, z=59.50], EntityBat['Bat'/1533, l='MpServer', x=113.44, y=34.18, z=69.84], EntityBat['Bat'/1791, l='MpServer', x=99.19, y=44.16, z=27.38]]
	Retry entities: 0 total; []
	Server brand: fml,forge
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:420)
	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2738)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:418)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Mac OS X (x86_64) version 10.13.3
	Java Version: 1.8.0_171, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 2868864816 bytes (2735 MB) / 3966763008 bytes (3783 MB) up to 7524057088 bytes (7175 MB)
	JVM Flags: 3 total; -Xmx8072m -Xms2048m -XX:PermSize=256m
	IntCache: cache: 0, tcache: 0, allocated: 13, tallocated: 95
	FML: MCP 9.42 Powered by Forge 14.23.3.2673 6 mods loaded, 6 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State     | ID         | Version      | Source                           | Signature                                |
	|:--------- |:---------- |:------------ |:-------------------------------- |:---------------------------------------- |
	| UCHIJAAAA | minecraft  | 1.12.2       | minecraft.jar                    | None                                     |
	| UCHIJAAAA | mcp        | 9.42         | minecraft.jar                    | None                                     |
	| UCHIJAAAA | FML        | 8.0.99.99    | forge-1.12.2-14.23.3.2673.jar    | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | forge      | 14.23.3.2673 | forge-1.12.2-14.23.3.2673.jar    | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | baubles    | 1.5.2        | Baubles-1.12-1.5.2.jar           | None                                     |
	| UCHIJAAAA | thaumcraft | 6.1.BETA10   | Thaumcraft-1.12.2-6.1.BETA10.jar | None                                     |

	Loaded coremods (and transformers): 
	GL info: ' Vendor: 'ATI Technologies Inc.' Version: '2.1 ATI-1.60.26' Renderer: 'AMD Radeon Pro 560 OpenGL Engine'
	Launched Version: forge-14.23.3.2673
	LWJGL: 2.9.2
	OpenGL: AMD Radeon Pro 560 OpenGL Engine GL version 2.1 ATI-1.60.26, ATI Technologies Inc.
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because ARB_framebuffer_object is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fml,forge'
	Type: Client (map_client.txt)
	Resource Packs: Isabella II × SuikaCraft 1.12
	Current Language: English (US)
	Profiler Position: N/A (disabled)
	CPU: 8x Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
```

**Comment by wamc2017** [comment]
Update to beta 11... ;)

**Comment by TimeConqueror** [comment]
**Beta 11.** 
The same thing happened when I checked the recipe for Enchanted Fabric in research "Discovering Infusion".  But my crash-report says: "java.lang.ArrayIndexOutOfBoundsException: 1":
[crash-2018-04-26_17.07.40-client.txt](https://github.com/Azanor/thaumcraft-beta/files/1951537/crash-2018-04-26_17.07.40-client.txt)

**Comment by codewarrior0** [comment]
Beta 11, also got AIOOB exception when viewing recipe for Essentia Centrifuge:

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
	at thaumcraft.client.gui.GuiResearchPage.drawStackAt(GuiResearchPage.java:2035) ~[GuiResearchPage.class:?]
	at thaumcraft.client.gui.GuiResearchPage.drawArcaneCraftingPage(GuiResearchPage.java:1432) ~[GuiResearchPage.class:?]
	at thaumcraft.client.gui.GuiResearchPage.drawRecipe(GuiResearchPage.java:1037) ~[GuiResearchPage.class:?]
	at thaumcraft.client.gui.GuiResearchPage.drawPage(GuiResearchPage.java:465) ~[GuiResearchPage.class:?]
	at thaumcraft.client.gui.GuiResearchPage.genResearchBackground(GuiResearchPage.java:321) ~[GuiResearchPage.class:?]
	at thaumcraft.client.gui.GuiResearchPage.func_73863_a(GuiResearchPage.java:261) ~[GuiResearchPage.class:?]
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:368) ~[ForgeHooksClient.class:?]
	at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1124) ~[buq.class:?]
	... 17 more
```

**Comment by hanleybrand** [comment]
@wamc2017  -- Yeah, unfortunately I had already made the Thaumometer when I updated to beta 11, so I couldn't confirm that had specifically fixed that issue. 

I had discovered that the crash only happened when I clicked the recipe at the bottom of the page, but the bookmark on the right side of the book showed the recipe fine. 

Unfortunately, further recipes (Nitor) crash with the outofboundsexception @TimeConqueror  mentions above... ☹️ But I guess that's another story.
